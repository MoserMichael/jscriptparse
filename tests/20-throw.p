    # checks formatting of error messages

    def rec(num) {
        if num == 0 {
          throw "throwing an error string"
        }
        return rec(num - 1)
    }        
    
    rec(5)
