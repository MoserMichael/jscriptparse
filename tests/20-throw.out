Error: "throwing an error string"
#(tests/20-throw.p:13)           throw "throwing an error string"
                     |...........^
#(tests/20-throw.p:12)     def rec(num) {
                     |.....^
#(tests/20-throw.p:11)         return rec(num - 1)
                     |................^
#(tests/20-throw.p:10)     def rec(num) {
                     |.....^
#(tests/20-throw.p:9)         return rec(num - 1)
                    |................^
#(tests/20-throw.p:8)     def rec(num) {
                    |.....^
#(tests/20-throw.p:7)         return rec(num - 1)
                    |................^
#(tests/20-throw.p:6)     def rec(num) {
                    |.....^
#(tests/20-throw.p:5)         return rec(num - 1)
                    |................^
#(tests/20-throw.p:4)     def rec(num) {
                    |.....^
#(tests/20-throw.p:3)         return rec(num - 1)
                    |................^
#(tests/20-throw.p:2)     def rec(num) {
                    |.....^
#(tests/20-throw.p:1)     rec(5)
                    |.....^
